### PR TITLE
[Cloud Tasks] Add HTTP Queue Sample

### PR DESCRIPTION
**Pull request by averikitsch**
Test projects will have to be granted permission for HTTP targets or wait until whitelisting is off.

### CODE REVIEW

File: appengine/cloud-tasks/create_http_task.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+# @param [String] url The full path to sent the task request to.
+# @param [String] payload The request body of your task.
+# @param [Integer] seconds The delay, in seconds, to process your task.
+def create_http_task project_id, location_id, queue_id, url, payload: nil, seconds: nil
+
+  # Instantiates a client.
+  client = Google::Cloud::Tasks.new
+
+  # Construct the fully qualified queue name.
+  parent = Google::Cloud::Tasks::V2beta3::CloudTasksClient.queue_path project_id, location_id, queue_id
```
Comment (TheRoyalTnetennba) [inline appengine/cloud-tasks/create_http_task.rb]: This PR reminded me, so I opened and released https://github.com/googleapis/google-cloud-ruby/pull/3034. `Google::Cloud::Tasks::V2beta3::CloudTasksClient.queue_path` can now also be accessed via `client.queue_path`

**Comment by averikitsch** [comment]
Ping @TheRoyalTnetennba, do you mind reviewing today or tomorrow?

File: appengine/cloud-tasks/create_task.rb [hunk 2]
```diff
@@ -23,10 +23,10 @@
 # @param [Integer] seconds The delay, in seconds, to process your task.
 def create_task project_id, location_id, queue_id, payload: nil, seconds: nil
   # Instantiates a client.
-  cloud_tasks = Google::Cloud::Tasks.new
+  client = Google::Cloud::Tasks.new
```
Comment (TheRoyalTnetennba) [inline appengine/cloud-tasks/create_task.rb]: What was the rationale for this change? I'm somewhat partial to `cloud_tasks`

File: appengine/cloud-tasks/spec/task_spec.rb [hunk 3]
```diff
@@ -16,23 +16,25 @@
 require "rspec"
 require "rack/test"
 require "google/cloud/tasks"
+require 'cgi'
```
Comment (TheRoyalTnetennba) [inline appengine/cloud-tasks/spec/task_spec.rb]: bit nit-picky, but could this be double quotes?

Comment (averikitsch) [reply appengine/cloud-tasks/spec/task_spec.rb]: done.

File: appengine/cloud-tasks/create_task.rb (hunk 2 above)
Comment (averikitsch) [reply appengine/cloud-tasks/create_task.rb]: I thought it fit the rubric better by naming it client. Would you prefer `cloud_tasks_client`?

Comment (TheRoyalTnetennba) [reply appengine/cloud-tasks/create_task.rb]: Yeah. Up to you. We haven't been super consistent on this.

**Comment by averikitsch** [comment]
To remain consistent with the other languages, the http target sample is moving to a folder not under appengine.

File: tasks/create_http_task.rb [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      queue_id,
+      url,
+      payload: payload,
+      seconds: seconds
+    )
+  else
+    puts <<~USAGE
+      Usage: ruby create_task.rb <GOOGLE_CLOUD_PROJECT> <LOCATION_ID> <QUEUE_ID> <payload> <seconds>
+
+      Environment variables:
```
Comment (TheRoyalTnetennba) [inline tasks/create_http_task.rb]: Should QUEUE_ID and LOCATION_ID be here?

File: tasks/create_http_task.rb [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  response = client.create_task parent, task
+
+  if response.name
+    puts "Created task #{response.name}"
+  end
+end
+# [END cloud_tasks_appengine_create_task]
+
+if $PROGRAM_NAME == __FILE__
+  project_id  = ARGV.shift
```
Comment (TheRoyalTnetennba) [inline tasks/create_http_task.rb]: Elsewhere in the repo, project_id is grabbed from ENV rather than as an argument.

Comment (TheRoyalTnetennba) [reply tasks/create_http_task.rb]: I.E. https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/13b453bd54baf7eed1e1d40bd6a45e9cf007f91a/storage/iam.rb#L73

File: tasks/README.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
+# Ruby Google Cloud Tasks sample for Google App Engine
+
+This sample application shows how to use [Google Cloud Tasks](https://cloud.google.com/cloud-tasks/)
+on Google App Engine [flexible environment][appengine-flex].
+
+App Engine queues push tasks to an App Engine HTTP target. This directory
+contains both the App Engine app to deploy, as well as the snippets to run
+locally to push tasks to it, which could also be called on App Engine.
+
+`create_task.rb` is a simple command-line program to create tasks to be pushed to
```
Comment (TheRoyalTnetennba) [inline tasks/README.md]: This README may need to be split up between the tasks/ and the appengine/cloud-tasks/ one

Comment (averikitsch) [reply tasks/README.md]: done. Thank you for catching this.

**Comment by averikitsch** [comment]
ping @TheRoyalTnetennba (Sorry to bug you again, but trying to get this done by the EOW)

File: tasks/README.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+example url to send requests to the App Engine task handler:
+
+```
+export URL=https://${GOOGLE_CLOUD_PROJECT}.appspot.com/log_payload
+```
+Running the sample will create a task and send the task to the specific URL
+endpoint, with a payload specified:
+
+```
+ruby create_http_task.rb $URL
```
Comment (TheRoyalTnetennba) [inline tasks/README.md]: `ruby create_http_task.rb $LOCATION_ID $QUEUE_ID $URL`

**Pull request merged**